src/v/cluster/tx_gateway_frontend.cc

```diff
@@ -141,9 +141,11 @@ void tx_gateway_frontend::start_expire_timer() {
 }
 
 ss::future<> tx_gateway_frontend::stop() {
+    vlog(txlog.debug, "Asked to stop tx coordinator");
```

Comment:
I think the real culprit is this commit https://github.com/redpanda-data/redpanda/commit/7e27f29756b67a77fc13a6511159178fb641e719 .. it changed the order in which we call stop(). 

After the change we stop the partitions first (which stops tm_stm) and then stop tx_gateway_frontend service... gating here is nice but until we fix that order we don't actually fix this issue? Looking at the commit message there, it seems like frontend is not stopping because of an in progress replicate in the underlying partition, so we seem to have some sort of circular dependency.

Reply:
it looks like it fixed it

Reply:
As we chatted offline, I'm not entirely sure we are fixing the issue here, we are probably just getting lucky.. gating in general seems like a harmless change, consider updating the PR title as we are not sure we are fixing UAF here, also mention the problematic commit reference in the PR title if ok.

Reply:
Agree, decided to go ahead and merge it: at least it improves logging so we'll get more data points if the issue isn't fixed. It would be nice to know for sure if the fix works but I wasn't be able to reproduce it. So far the best bet is to merge it and reopen if it happens again.